### PR TITLE
[IAppConfig] new tests

### PR DESCRIPTION
**Pull request by ArtificialOwl**
- moving previous test into another test file
- new tests for the new API IAppConfig
- fix the search on app id (prefix)
- remove useless check on key, redundant with `isLazy()`

### CODE REVIEW

**Review by come-nc — APPROVED** [review]
Do we have to keep a deprecated test class around? Simply remove it, no?

**Comment by ArtificialOwl** [comment]
> Do we have to keep a deprecated test class around? Simply remove it, no?

was thinking to keep them for a while to ensure compatibility of the deprecated methods

File: tests/lib/AppConfigTest.php [hunk 1]
```diff
@@ -22,344 +39,1325 @@
...
+	/**
+	 * @param bool $preLoading TRUE will preload the 'fast' cache, which is the normal behavior of usual
+	 *     IAppConfig
+	 *
+	 * @return IAppConfig
+	 */
+	private function generateAppConfig(bool $preLoading = true): IAppConfig {
+		/** @var AppConfig $config */
+		$config = new \OC\AppConfig(
+			\OC::$server->get(IDBConnection::class), \OC::$server->get(LoggerInterface::class)
```
Comment (nickvergessen) [inline tests/lib/AppConfigTest.php:190]:
```suggestion
			$this->connection, $this->logger
```

File: lib/private/AppConfig.php [hunk 2]
```diff
@@ -839,10 +844,6 @@ public function updateType(string $app, string $key, int $type = self::VALUE_MIX
 		$this->loadConfigAll();
 		$lazy = $this->isLazy($app, $key);
 
-		if (!$this->hasKey($app, $key, $lazy)) {
```
[the remark this answers is not in the document]
Comment (ArtificialOwl) [reply lib/private/AppConfig.php:842]: useless call to `hasKey()` as it is already done in `isLazy()`

[the remark this answers is not in the document]
Comment (ArtificialOwl) [reply lib/private/AppConfig.php:842]: @skjnldsv done

File: lib/private/AppConfig.php [hunk 3]
```diff
@@ -210,12 +210,17 @@ public function isLazy(string $app, string $key): bool {
 	 * @return array<string, string> [configKey => configValue]
 	 * @since 29.0.0
 	 */
-	public function getAllValues(string $app, string $key = '', bool $filtered = false): array {
-		$this->assertParams($app, $key);
+	public function getAllValues(string $app, string $prefix = '', bool $filtered = false): array {
```
Comment (emoral435) [inline lib/private/AppConfig.php:213]: nitpick: the functions above description still has  `@param string $key`, so ->
```js
@param string $prefix
```

**Review by emoral435 — APPROVED** [review]
Aside from small nitpick, and me agreeing with @come-nc that we do not really need to keep the old tests, everything looks good to me. But I do agree that we really should remove it, especially if it is not being used.

**Comment by ArtificialOwl** [comment]
- fixed phpdoc
- removing deprecated tests

**Comment by ArtificialOwl** [comment]
_(failing check not related)_

**Pull request merged**